From 701420d7e19bebfba1fe67163eabef0365a63c94 Mon Sep 17 00:00:00 2001
From: Andrew Bartlett <abartlet@samba.org>
Date: Mon, 30 Jan 2017 09:36:31 -0500
Subject: [PATCH 13/15] autobuild: Add nopython environment to test
 --disable-python  builds (but without tests)

This ensures we keep this option building as we extend our use of python.

The rule is that new features and changes to existing features that
require python are most welcome, they just need to be disabled for the
minimalistic targets we still ecourage Samba on, that typically just
want smbd
---
 .travis.yml         |  1 +
 script/autobuild.py | 18 +++++++++++++++++-
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/.travis.yml b/.travis.yml
index 483ad501798..ce0e745548b 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -12,6 +12,7 @@ env:
   - TASK=samba-libs
   - TASK=samba-static
   - TASK=samba-o3
+  - TASK=samba-nopython
   - TASK=ldb
   - TASK=tdb
   - TASK=talloc
diff --git a/script/autobuild.py b/script/autobuild.py
index e46869c0407..27deec472df 100755
--- a/script/autobuild.py
+++ b/script/autobuild.py
@@ -32,6 +32,7 @@ builddirs = {
     "samba-static"  : ".",
     "samba-test-only"  : ".",
     "samba-systemkrb5"  : ".",
+    "samba-nopython"  : ".",
     "ldb"     : "lib/ldb",
     "tdb"     : "lib/tdb",
     "talloc"  : "lib/talloc",
@@ -43,7 +44,7 @@ builddirs = {
     "retry"   : "."
     }
 
-defaulttasks = [ "ctdb", "samba", "samba-xc", "samba-o3", "samba-ctdb", "samba-libs", "samba-static", "samba-systemkrb5", "ldb", "tdb", "talloc", "replace", "tevent", "pidl" ]
+defaulttasks = [ "ctdb", "samba", "samba-xc", "samba-o3", "samba-ctdb", "samba-libs", "samba-static", "samba-systemkrb5", "samba-nopython", "ldb", "tdb", "talloc", "replace", "tevent", "pidl" ]
 
 if os.environ.get("AUTOBUILD_SKIP_SAMBA_O3", "0") == "1":
     defaulttasks.remove("samba-o3")
@@ -177,6 +178,21 @@ tasks = {
                       ("clean", "make clean", "text/plain")
                       ],
 
+    # Test Samba without python still builds.  When this test fails
+    # due to more use of Python, the expectations is that the newly
+    # failing part of the code should be disabled when
+    # --disable-python is set (rather than major work being done to
+    # support this environment).  The target here is for vendors
+    # shipping a minimal smbd.
+    "samba-nopython" : [
+                      ("random-sleep", "script/random-sleep.sh 60 600", "text/plain"),
+                      ("configure", "./configure.developer --picky-developer ${PREFIX} --with-profiling-data --disable-python --without-ad-dc", "text/plain"),
+                      ("make", "make -j", "text/plain"),
+                      ("install", "make install", "text/plain"),
+                      ("check-clean-tree", "script/clean-source-tree.sh", "text/plain"),
+                      ("clean", "make clean", "text/plain")
+                      ],
+
 
 
     "ldb" : [
-- 
2.13.0

